cmake/Modules/FindMagic_EP.cmake

```diff
@@ -0,0 +1,150 @@
+#
+# FindMagic_EP.cmake
+#
+# The MIT License
+#
+# Copyright (c) 2018-2021 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# Finds the Magic library, installing with an ExternalProject as necessary.
+# This module defines:
+#   - libmagic_INCLUDE_DIR, directory containing headers
+#   - libmagic_LIBRARIES, the Magic library path
+#   - libmagic_FOUND, whether Magic has been found
+#   - The libmagic imported target
+
+# Include some common helper functions.
+include(TileDBCommon)
+
+# Search the path set during the superbuild for the EP.
+set(LIBMAGIC_PATHS ${TILEDB_EP_INSTALL_PREFIX})
+
+if(TILEDB_LIBMAGIC_EP_BUILT)
+  find_package(libmagic PATHS ${TILEDB_EP_INSTALL_PREFIX} ${TILEDB_DEPS_NO_DEFAULT_PATH})
+endif()
+
+if (TILEDB_LIBMAGIC_EP_BUILT)
+  find_path(libmagic_INCLUDE_DIR
+    NAMES magic.h
+    PATHS ${LIBMAGIC_PATHS}
+    PATH_SUFFIXES include
+    ${NO_DEFAULT_PATH}
+  )
+
+  if (NOT libmagic_INCLUDE_DIR)
+    find_path(libmagic_INCLUDE_DIR
+      NAMES file/file.h
+      PATHS ${LIBMAGIC_PATHS}
+      PATH_SUFFIXES include
+      ${NO_DEFAULT_PATH}
+    )
+  endif()
+
+  # Link statically if installed with the EP.
+  find_library(libmagic_LIBRARIES
+    libmagic
+    PATHS ${LIBMAGIC_PATHS}
+    PATH_SUFFIXES lib a
+    #${TILEDB_DEPS_NO_DEFAULT_PATH}
+    ${NO_DEFAULT_PATH}
+  )
+
+  find_library(libmagic_pcre2posix_LIBRARIES
```

Comment:
Not clear why we have two find invocations with (I assume) the same underlying library. Shouldn't this be unnecessary now that we are building libmagic unconditonally?

If this is unavoidable, could `HINTS` and a single `find_library` invocation.